pymatgen/analysis/fragmenter.py

```diff
@@ -78,8 +82,8 @@ def __init__(
         if edges is None:
             self.mol_graph = MoleculeGraph.with_local_env_strategy(molecule, OpenBabelNN())
         else:
-            edges = {(e[0], e[1]): None for e in edges}
-            self.mol_graph = MoleculeGraph.with_edges(molecule, edges)
+            _edges = {(edge[0], edge[1]): None for edge in edges}
```

Comment:
Can you please give me a hand on this `mypy` error:

```python
pymatgen/analysis/fragmenter.py:86: error: Argument 2 to "with_edges" of "MoleculeGraph" has incompatible type "dict[tuple[int, int], None]"; expected "dict[tuple[int, int], Optional[dict[Any, Any]]]"  [arg-type]
pymatgen/analysis/fragmenter.py:86: note: "Dict" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
pymatgen/analysis/fragmenter.py:86: note: Consider using "Mapping" instead, which is covariant in the value type
```

I'm not sure what is the reason. Thanks a lot.

Reply:
@janosh

Reply:
No worries. Any help would be appreciated!